### PR TITLE
fixing ansible vars issue

### PR DESCRIPTION
**Pull request by JAGANNATH-OPTIT**
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

### Type of change

Please choose appropriate options.

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Enhancement (additive changes to improve performance)
- [ ] This change requires a documentation update

### How Has This Been Tested?

Please describe the tests that you ran to verify your changes in the below checkboxes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Ran Test A
- [ ] Ran Test B

**Test Configuration**:
* Software versions:
* Hardware versions:

### Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: ansible/roles/ml-analytics-service/defaults/main.yml [hunk 1]
```diff
@@ -37,13 +37,18 @@ ml_analytics_kafka_survey_druid_topic_name: "{{ env}}.ml.survey.druid"
 ml_analytics_observation_log_folder_path: "{{ WORKDIR }}/logs/observation"
 ml_analytics_project_log_folder_path: "{{ WORKDIR }}/logs/project"
 ml_analytics_survey_log_folder_path: "{{ WORKDIR }}/logs/survey"
-ml_analytics_es_url: "{{ml_survey_elasticsearch_host_url | default("http://"+(groups['es7'][0]+':9200'))}}"
-ml_analytics_azure_account_name: "{{ml_analytics_azure_account_name}}"
-ml_analytics_azure_container_name: "{{ml_analytics_azure_container_name}}"
+ml_analytics_es_url: "{{ml_survey_elasticsearch_host_url | default(groups['es'][0]+':9200')}}"
```
Comment (keshavprasadms) [inline ansible/roles/ml-analytics-service/defaults/main.yml:40]: Are we referring to ES version 6 or 7?

File: ansible/roles/ml-analytics-service/defaults/main.yml [hunk 2]
```diff
@@ -37,13 +37,18 @@ ml_analytics_kafka_survey_druid_topic_name: "{{ env}}.ml.survey.druid"
...
+ml_analytics_azure_account_name: "{{ml_analytics_account_name | default('sunbird_public_storage_account_name') }}"
+ml_analytics_azure_container_name: "{{ml_analytics_container_name | default('samiksha')}}"
 ml_analytics_observation_azure_blob_path: "observation/status/"
 ml_analytics_project_azure_blob_path: "reports/"
-ml_analytics_redis_host: "{{ml_redis_host | default(groups['dp-redis'][0])}}"
-ml_analytics_redis_port: "6381"
-ml_analytics_redis_db_name: "16"
+ml_analytics_redis_host: "{% if metadata2_redis_host is defined %}{{ metadata2_redis_host }}{% else %}{{ redis_host }}{% endif %}"
```
Comment (keshavprasadms) [inline ansible/roles/ml-analytics-service/defaults/main.yml:45]: Do we know something called `redis_host` is defined in defaults / all.yml?

File: ansible/roles/ml-analytics-service/templates/config.j2 [hunk 3]
```diff
@@ -28,7 +28,7 @@ evidence_base_url = {{ ml_analytics_evidence_base_url }}
 # Mongo  url
 #---------------
 
-url = mongodb://{{ ml_analytics_mongodb_url }}
+mongo_url = mongodb://{{ ml_analytics_mongodb_url }}
```
Comment (keshavprasadms) [inline ansible/roles/ml-analytics-service/templates/config.j2:31]: What is the default value?

Comment (JAGANNATH-OPTIT) [reply ansible/roles/ml-analytics-service/templates/config.j2:31]: This is configuration file for service. In code it is referring to mongo_url so updated the syntax which missed here. 
https://github.com/Sunbird-Ed/ml-analytics-service/blob/9be74f5a849c1d5d82408d12a00a6b8e777c8e0d/observations/py_observation_evidence_streaming.py#L67

File: ansible/roles/ml-analytics-service/defaults/main.yml [hunk 4]
```diff
@@ -37,13 +37,17 @@ ml_analytics_kafka_survey_druid_topic_name: "{{ env}}.ml.survey.druid"
...
-ml_analytics_azure_account_name: "{{ml_analytics_azure_account_name}}"
-ml_analytics_azure_container_name: "{{ml_analytics_azure_container_name}}"
+ml_analytics_azure_account_name: "{{ ml_cloud_config.AZURE_ACCOUNT_NAME | default('sunbird_public_storage_account_name') }}"
+ml_analytics_azure_container_name: "{{ ml_cloud_config.AZURE_ACCOUNT_KEY | default('samiksha')}}"
 ml_analytics_observation_azure_blob_path: "observation/status/"
 ml_analytics_project_azure_blob_path: "reports/"
-ml_analytics_redis_host: "{{ml_redis_host | default(groups['dp-redis'][0])}}"
-ml_analytics_redis_port: "6381"
-ml_analytics_redis_db_name: "16"
+ml_analytics_redis_host: "{{ metadata_redis_host }}"
```
Comment (JAGANNATH-OPTIT) [inline ansible/roles/ml-analytics-service/defaults/main.yml:44]: We are referring this host here https://github.com/JAGANNATH-OPTIT/sunbird-devops/blob/0223f398b45784aeb3bd9c0705de9dd34d379dfa/ansible/roles/stack-sunbird/templates/analytics_api_service.conf#L210

**Review by saiakhil46 — APPROVED** [review]

**Review by keshavprasadms — APPROVED** [review]

**Pull request merged**